### ISSUE TITLE
Doubled legacy tracks and moves when with assistant GM.

### ISSUE THREAD

**Issue by bblonski** [DiceT/starforged#4]
I tried playing co-op with another player assigned with the Assistant GM role. When creating new character actors, there were duplicate Legacy Tracks and Moves. E.g. The legacies tab showed: Quests, Discoveries, Bonds, Quests, Discoveries, Bonds, Background Vow, Background Vow. Changing the 2nd players assigned role to Trusted Player resolved the issue, so new character actors would be created properly with one set of Legacies and Moves. 

Both players need to be logged in for the duplication to occur. Having both players as full Gamemaster roles does not duplicate the Quests, Discoveries, and Bonds legacy tracks, but still duplicates the moves and background vows.

**Comment by DiceT** [comment]
The multiplayer aspect eludes me at the moment. I will be investigating this further for an upcoming patch. It won't be the next patch, but I have it on my TODO list now.

**Comment by bblonski** [comment]
I looked around the code and I think I figured out the issue. You implemented adding the legacy tracks and moves on the createActor hook, and hooks are executed per player logged in. If 2 players are Gamemasters, they both have permission to add and they end up duplicated.

Instead they should probably be implemented in the constructor of StarforgedActor, or something similar. I haven't tried to move the code over yet, so I don't know if you can get to the same game data. But hopefully it's mostly just a matter of moving the code over.

**Comment by DiceT** [comment]
I don't believe I have any means to test this, but I implemented your suggestion. It likely requires you to Delete All on your Moves folder and then refresh so the Moves are repopulated.
